kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -69,31 +142,54 @@ extension [A, S](effect: A < S)
         effect.map(v => loop(v))
     end repeatWhile
 
+    /** Performs this computation repeatedly while the given condition holds.
+      *
+      * @param fn
+      *   The condition to check after each iteration taking the current value and the number of iterations so far, and returning a tuple
+      *   with the condition and the duration to sleep between iterations
+      * @return
+      *   A computation that produces the result of the last successful execution before the condition becomes false
+      */
     def repeatWhile[S1](fn: (A, Int) => (Boolean, Duration) < S1)(using
-        Flat[A < S],
```

Comment:
This was still using an old encoding of `Flat`, making calls to the method not compile.

Reply:
Did I not have tests for the repeat methods?

Reply:
Yeah, I think so. They were missing the effect on each loop iteration.